src/components/SetupApplication/SetupApplicationForm.tsx

```diff
@@ -54,7 +54,7 @@ export const SetupApplicationForm: FC<Props> = ({
 
     const [formState, { text }] = useFormState(initialState);
 
-    const checkForErrors = () => {
+    useEffect(() => {
```

Comment:
Rather than the `validSoFar` stuff, could you just setValidForm(false) as soon as it's invalid, and return early?